File: R_note/inc_menu/example/package/mypkg/R/call_c.r
# File name: call_c.r

# dyn.load("call_c.so")
# For Windows will like this
# dyn.load("C:/Windows/Desktop/call_c.dll")

# symbol.C("hello")
# is.loaded(symbol.C("hello"))

# a <- 1 : 9

test.c <- function(a) {
  b <- a
  d <- vector(mode = "numeric", length = length(a))
  e <- vector(mode = "numeric", length = 1)
  ret <- .C("hello", m = length(a), a = as.double(a),
           b = as.double(b), d = as.double(d), e = as.double(e))
  ret
}

# test.c(a)

# dyn.unload("call_c.so")
# For Windows will like this
# dyn.unload("C:/Windows/Desktop/call_c.dll")

File: R_note/inc_menu/example/package/mypkg/src/call_c.c
/*
  File name: call_c.c
  For dynamical load compile by gcc.
  SHELL> gcc -c call_c.c ; gcc -shared -o call_c.so call_c.o
*/

void hello(int *m, double *a, double *b, double *c, double *d){
  int i;

  *d = 0;
  for(i = 0; i < *m; i++){
    c[i] = a[i] + b[i];
    *d += c[i];
    a[i] = 0;
  }
}

/* Output is a shared library "call_c.so" can called by R. */
